ui_tests/pages/inventory_page.py

```diff
@@ -0,0 +1,336 @@
+            host_error = driver.find_element(By.XPATH, host_error_xpath)
+            return host_error.text
+        except selenium.common.exceptions.NoSuchElementException:
+            return None
+
+    def get_community_string_missing_error(self):
+        logger.info(f"Get community string missing error")
+        community_error_xpath = f"//p[@data-test='sc4snmp:community-error']"
+        try:
+            community_error = driver.find_element(By.XPATH, community_error_xpath)
+            return community_error.text
+        except selenium.common.exceptions.NoSuchElementException:
+            return None
+
+    def get_walk_invalid_value_error(self):
+        logger.info(f"Get walk interval invalid value error")
+        community_error_xpath = f"//p[@data-test='sc4snmp:walk-interval-error']"
+        try:
+            community_error = driver.find_element(By.XPATH, community_error_xpath)
+            return community_error.text
+        except selenium.common.exceptions.NoSuchElementException:
+            return None
+
+    def edit_device_port(self, port):
+        logger.info(f"set/edit inventory device port: {port}")
+        device_port_field_xpath = (
+            "//div[@data-test='sc4snmp:form:port-input']//span//input"
+        )
+        port_field = driver.find_element(By.XPATH, device_port_field_xpath)
+        helper.clear_input(port_field)
+        port_field.send_keys(port)
+
+    def select_snmp_version(self, snmp_version):
+        logger.info(f"set device snmp version: {snmp_version}")
+        options = {
+            "1": "//button[@data-test='sc4snmp:form:version-1']",
+            "2c": "//button[@data-test='sc4snmp:form:version-2c']",
+            "3": "//button[@data-test='sc4snmp:form:version-3']",
+        }
+        snmp_version_expander_xpath = (
+            "//button[@data-test='sc4snmp:form:select-version']"
+        )
+        expander = driver.find_element(By.XPATH, snmp_version_expander_xpath)
+        expander.click()
+        time.sleep(1)
+        option = driver.find_element(By.XPATH, options[snmp_version])
+        option.click()
+
+    def set_secret(self, secret, edit=False):
+        logger.info(f"set inventory device secret: {secret}")
+        secret_field_xpath = (
+            "//div[@data-test='sc4snmp:form:secret-input']//span//input"
+        )
+        secret_field = driver.find_element(By.XPATH, secret_field_xpath)
+        if edit:
+            helper.clear_input(secret_field)
+        secret_field.send_keys(secret)
+
+    def set_security_engine(self, security_engine, edit=False):
+        logger.info(f"set inventory device security engine: {security_engine}")
+        sec_engine_field_xpath = (
+            "//div[@data-test='sc4snmp:form:security-engine-input']//span//input"
+        )
+        sec_engine = driver.find_element(By.XPATH, sec_engine_field_xpath)
+        if edit:
+            helper.clear_input(sec_engine)
+        sec_engine.send_keys(security_engine)
```

Comment:
Similar set functions to ones before

Reply:
3 similar functions merged into _set_inventory_field(self, field, value, edit=False)